packages/legend-extension-dsl-data-space/src/components/query-builder/DataSpaceQueryBuilder.tsx

```diff
@@ -225,10 +228,55 @@ const DataSpaceQueryBuilderSetupPanelContent = observer(
       );
     }, [queryBuilderState, applicationStore]);
 
+    const copyDataSpaceLinkToClipboard = (): void => {
+      const nonNullableProject = guaranteeNonNullable(
+        project,
+        'Unable to copy data space link to clipboard because project is null',
+      );
+      const dataSpace = queryBuilderState.dataSpace;
+      const executionContext = queryBuilderState.executionContext;
+      const runtimePath =
+        queryBuilderState.executionContextState.runtimeValue instanceof
+        RuntimePointer
+          ? queryBuilderState.executionContextState.runtimeValue
+              .packageableRuntime.value.path
+          : undefined;
+      const route =
+        applicationStore.navigationService.navigator.generateAddress(
+          generateDataSpaceQueryCreatorRoute(
+            nonNullableProject.groupId,
+            nonNullableProject.artifactId,
+            nonNullableProject.versionId,
+            dataSpace.path,
+            executionContext.name,
+            runtimePath,
+            queryBuilderState.class?.path,
+          ),
+        );
+
+      navigator.clipboard
+        .writeText(route)
+        .catch(() =>
+          applicationStore.notificationService.notifyError(
+            'Error copying data space link to clipboard',
+          ),
+        );
+
+      applicationStore.notificationService.notifySuccess(
+        'Copied data space link to clipboard',
+      );
+    };
+
     return (
       <div className="query-builder__setup__config-group">
         <PanelHeader title="properties">
           <PanelHeaderActions>
+            <PanelHeaderActionItem
+              title="copy data space link to clipboard"
```

Comment:
do you think if we should disable when there is no dataspace selected in the dataspace dropdown?

Reply:
Done!